### PR TITLE
set plugin group according to type if possible, fixes #390

### LINKED ISSUE #390: option -t not working on develop
I cloned a fresh repo, and tried to run following command.
```bash
./owtf.py -t active http://127.0.0.1

 _____ _ _ _ _____ _____
|     | | | |_   _|   __|
|  |  | | | | | | |   __|
|_____|_____| |_| |__|

        @owtfp
    http://owtf.org
    
Current Path: ./owtf.py
Syntax: owtf.py [ options ] <target1 target2 target3 ..> where target can be: <target URL / hostname / IP>
                    NOTE: targets can also be provided via a text file


Examples:

Run all web plugins:                         owtf.py http://my.website.com
Run only passive + semi_passive plugins:             owtf.py -t quiet http://my.website.com
Run only active plugins:                     owtf.py -t active http://my.website.com

Run all plugins except 'OWASP-CM-001: Testing_for_SSL-TLS': owtf.py -e 'OWASP-CM-001' http://my.website.com
Run all plugins except 'OWASP-CM-001: Testing_for_SSL-TLS': owtf.py -e 'Testing_for_SSL-TLS' http://my.website.com

Run only 'OWASP-CM-001: Testing_for_SSL-TLS':             owtf.py -o 'OWASP-CM-001' http://my.website.com
Run only 'OWASP-CM-001: Testing_for_SSL-TLS':             owtf.py -o 'Testing_for_SSL-TLS' http://my.website.com

Run only OWASP-IG-005 and OWASP-WU-VULN:             owtf.py -o 'OWASP-IG-005,OWASP-WU-VULN' http://my.website.com
Run using my resources file and proxy:             owtf.py -m r:/home/me/owtf_resources.cfg -x 127.0.0.1:8080 http://my.website.com

Run using TOR network:                    owtf.py -o OWTF-WVS-001 http://my.website.com --tor 127.0.0.1:9050:9051:password:1

Run Botnet-mode using miner:                    owtf.py -o OWTF-WVS-001 http://my.website.com -b miner

Run Botnet-mode using custom proxy list:                  owtf.py -o OWTF-WVS-001 http://my.website.com -b list:proxy_list_path.txt

ERROR: Invalid Plugin Type 'active' for Plugin Group 'None'. Valid Types: 
```

### PR DESCRIPTION
**Pull request by DarKnight--**
This PR sets the plugin group according to the plugin_type. This makes -t command functional.

## Description

Set the plugin_group according to type requested using `-t` option.

## Related Issue

#390 

## Reviewers

@delta24 

## How Has This Been Tested?

./owtf.py -t passive

## Screenshots (if appropriate):

screenshot from 2016-04-14 11 33 19

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [ ] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by delta24** [comment]
@DePierre is this what we want or in the cli_options.py?

File: owtf.py [hunk 1]
```diff
@@ -141,18 +142,25 @@ def process_options(user_args):
...
                 usage("Invalid port for Inbound Proxy")
 
-    plugin_types_for_group = ServiceLocator.get_component("db_plugin").GetTypesForGroup(plugin_group)
+    plugin_types_for_group = db_plugin.GetTypesForGroup(plugin_group)
     if arg.PluginType == 'all':
         arg.PluginType = plugin_types_for_group
     elif arg.PluginType == 'quiet':
         arg.PluginType = ['passive', 'semi_passive']
-        if plugin_group != 'web':
+        if plugin_group != None and plugin_group != 'web' :
```
Comment (DoomTaper) [inline owtf.py]: @DarKnight-- remove space before " : "

**Comment by DePierre** [comment]
@DarKnight-- @delta24 @7a Can we agree on the expected behavior here?

+ If the target is a hostname
  + If no -g and -t active
    + Run web/active plugins only
  + If no -g and -t passive
    + Run web/passive plugins only
  + If no -g and -t semi_passive
    + Run web/semi_passive plugins only
  + If no -g and -t quiet
    + Run web/passive and web/semi_passive plugins only
  + if no -g and -t bruteforce
    + Throw error "Invalid type for target"
+ If target is an IP
  + If no -g and -t active
    + Run net/active plugins only
  + If no -g and -t bruteforce
    + Run net/bruteforce plugins only
  + If no -g and -t quiet
    + Throw error "Invalid type for target"

**Comment by DarKnight--** [comment]
@DePierre @7a @delta24 
The current fix behaves in following way:
* if no -g
 * `-t <active or bruteforce>` => run the type with group based on target <hostname or ip>
 * `-t <any other type>` => runs the type with group based on type
* if -g is present
 * -t <type> => if type is present in the group, run those plugins
 * else throw an error

If group is not specified and that type is found in only one group then OWTF resolves the group. If multiple groups have those types of plugins (like active or bruteforce) the group is selected based on the target.

**Comment by 7a** [comment]
The understanding is correct, just bear in mind that, in the future we could have semi-passive + passive plugins for different plugin groups like "web". For example, for passive recon against a company, look at linkedin profiles, etc. that would have nothing to do with web, yet would be passive.

That said, ok from me :)

**Comment by DePierre** [comment]
@DarKnight-- I have merged part of your fix in https://github.com/owtf/owtf/commit/24aadb102a526ec2e8a62285e70fedde7d38797e. Thank you!

**Pull request closed without merging**